torch/csrc/jit/ir.cpp

```diff
@@ -897,20 +942,51 @@ const FunctionSchema* Node::maybeSchema() const {
   return nullptr;
 }
 
-const Operator& Node::getOperator() const {
+const Operator* Node::maybeOperator() const {
   if (!op_) {
-    op_ = &getOperatorFor(this);
+    const auto& candidates = getAllOperatorsFor(kind());
+    for (const auto& candidate : candidates) {
+      if (matches(candidate->schema())) {
+        op_ = candidate.get();
+        break;
+      }
+    }
   }
-  return *op_;
+  return op_;
 }
 
-const Operator* Node::maybeOperator() const {
-  if (!op_) {
-    if (auto op = findOperatorFor(this)) {
-      op_ = op.get();
+const Operator& Node::getOperator() const {
+  const Operator* maybe = maybeOperator();
+  if (maybe)
+    return *maybe;
+
+  auto er = script::ErrorReport(sourceRange());
```

Comment:
nit: matter of taste I guess, but I'd put the error builder in its own function rather than having it be like 80% of `getOperator()`. OK zooming out now